src/pq_flash_index.cpp

```diff
@@ -156,22 +180,12 @@ namespace diskann {
     diskann::cout << "Clearing scratch" << std::endl;
     assert(this->thread_data.size() == this->max_nthreads);
     while (this->thread_data.size() > 0) {
-      ThreadData<T> data = this->thread_data.pop();
-      while (data.scratch.sector_scratch == nullptr) {
+      auto data = this->thread_data.pop();
+      while (data->scratch.sector_scratch == nullptr) {
         this->thread_data.wait_for_push_notify();
         data = this->thread_data.pop();
       }
-      auto &scratch = data.scratch;
```

Comment:
Minor: these lines of code can be made into a method.

Reply:
LGTM.

Reply:
@hliu18, I am thinking of changing the pop method to return nullptr for data when all scratch spaces are taken. I can encapsulate as a method along with that change. can we delay your request until then or would you prefer to close before that?

Reply:
We can delay the suggested change. You can check in this PR ASAP so that Philip can update BANN package with your change.